pypsa/linopf.py

```diff
@@ -977,18 +981,33 @@ def msq_diff(n, s_nom_prev):
                     f"{lines_err}")
         return lines_err
 
-    iteration = 0
+    def save_optimal_capacities(n, iteration, status):
+        for c, attr in pd.Series(nominal_attrs)[n.branch_components].items():
+            n.df(c)[f'{attr}_opt_{iteration}'] = n.df(c)[f'{attr}_opt']
+        setattr(n, f"status_{iteration}", status)
+        setattr(n, f"objective_{iteration}", n.objective)
+        n.iteration = iteration
+
+
+    if track_iterations:
+        for c, attr in pd.Series(nominal_attrs)[n.branch_components].items():
+            n.df(c)[f'{attr}_opt_0'] = n.df(c)[f'{attr}']
+    iteration = 1
     kwargs['store_basis'] = True
     diff = msq_threshold
     while diff >= msq_threshold or iteration < min_iterations:
-        if iteration >= max_iterations:
+        if iteration > max_iterations:
             logger.info(f'Iteration {iteration} beyond max_iterations '
                         f'{max_iterations}. Stopping ...')
             break
 
         s_nom_prev = n.lines.s_nom_opt if iteration else n.lines.s_nom
         kwargs['warmstart'] = bool(iteration and ('basis_fn' in n.__dir__()))
-        network_lopf(n, snapshots, **kwargs)
+        status, termination_condition = network_lopf(n, snapshots, **kwargs)
+        assert status == 'ok', ('Optimization failed with status {status}'
+                                'and termination {termination_condition}')
```

Comment:
```suggestion
                                f'and termination {termination_condition}')
```
Isn't this necessary?